src/shared/components/FixedBottom.tsx

```diff
@@ -0,0 +1,27 @@
+'use client';
+
+import { useEffect } from 'react';
+import { useSetRecoilState, useResetRecoilState } from 'recoil';
+import { fixedBottomState } from '@/shared/atoms/fixedBottom';
+
+interface Props {
+  children: React.ReactNode;
+}
+
+const FixedBottom = ({ children }: Props) => {
+  const setFixedBottom = useSetRecoilState(fixedBottomState);
+  const resetFixedBottom = useResetRecoilState(fixedBottomState);
+
+  useEffect(() => {
+    setFixedBottom(children);
+    return resetFixedBottom;
+  }, [setFixedBottom, resetFixedBottom, children]);
+
+  return (
+    <div className="fixed bottom-0 left-1/2 -translate-x-1/2 z-[5001] max-w-[600px] w-full">
+      {children}
+    </div>
+  );
```

Comment:
```suggestion
return null
```

`<FixedBottomContainer />` 에서 fixedBottomState을 렌더링하는데,
여기서도 children을 렌더링하면, 2개가 겹처서 렌더링 되지 않나요?

Reply:
네네 맞아요
저도 처음에 `return null`로 해줬었는데,
상품 상세 페이지에서 새로고침하는 경우,
일단 먼저 `Footer`가 렌더링 되고 나중에 `FixedButton`의 useEffect callback function이 실행돼서
뒤늦게 구매하기 버튼으로 변경되더라구요..!

다른 방법을 고심해봤는데 좋은 아이디어가 떠오르지 않더라구요ㅠ
그래서 2개 겹쳐서 렌더링 되더라도 일단 위 이슈를 해결해야겠다 싶어서 저렇게 구현해뒀습니다.
혹시 더 좋은 방법이 있다면 의견 주시면 감사하겠습니다!🙏

Reply:
지금 다시 생각해보니깐, 
layout, page에서 중복 사용하면
layout의 FixedBottom이 렌더링 되는지, page의 FixedBottom이 렌더링 되는지
예측이 어려운 코드가 될 것 같아요.

아예 다른 로직이 필요할 것 같은데 어떻게 생각하시나요?

Reply:
useToast에서 style을 추가적으로 받는건 어떨까요?

Reply:
> layout, page에서 중복 사용하면
layout의 FixedBottom이 렌더링 되는지, page의 FixedBottom이 렌더링 되는지
예측이 어려운 코드가 될 것 같아요.

네네 그렇긴하죠

> useToast에서 style을 추가적으로 받는건 어떨까요?

`openToast`의 인자로 style을 받자는 말씀이실까요??

```js
openToast({ message: ..., action: ..., style: { bottom: 하단 요소 height + 10 } })
```

저는 위처럼 이해했는데 이게 맞을까요??

Reply:
```tsx
useToastNewver({style: { bottom: 하단 요소 height + 10 }});
```

이렇게요!

Reply:
그럼 저 style 값은 openToast 함수에서 setToast할 때 `{message, action, id, style}` 이렇게 저장해줘야 하는거죠??

Reply:
네네

Reply:
음.. 하단 고정 요소의 height 값을 일일이 적어줘야 한다는 불편함은 있지만 직관적이긴 한 것 같네요
이렇게 수정해 놓을게요!

Reply:
지성님~ 위 방법대로 하면 또 다른 이슈가 있네요ㅜㅜ
하나의 컴포넌트 또는 훅을 하단 고정 요소가 다른 곳에 재사용하는 경우
`useToastNewVer`의 style bottom 값으로 어떤 값을 넣어줘야 할지 애매한 상황이 생기네요

예를 들어 `AddAlarmPopup` 컴포넌트의 경우
상품 상세 페이지(하단 요소: 버튼)와 빵켓팅 페이지(하단 요소: 푸터)에 둘다 사용이 되는데
이때 토스트를 어느 높이를 기준으로 띄워야 할지 애매한 상황이 생기게 되네요

Reply:
![image](https://github.com/user-attachments/assets/666c398d-d8fc-4822-8cc8-552579cbe8fe)

아니면 레이아웃 자체를 이런식으로 바꿔서 토스트를 띄우는건 어떨까요?
잘 될지 모르겠는데, 검토 해주실 수 있을까요?

Reply:
오호 한번 해볼게요!

Reply:
지성님 생각해보니까 위 방법으로 해도 main의 height가 100vh에 하단 `Footer`의 height를 제외한 값이니까,
main 내부에서 `FixedBottom`에 의해 하단 고정 요소가 달라지면 어차피 토스트가 가려지긴 할 것 같아요

Reply:
FixedBottom 컴포넌트는 없애고,
ToastContainer를 main 하단에 배치하자는 말이었어요

Reply:
`FixedBottom`를 없앤다고 해도 어차피 상품 상세 페이지에서처럼 하단에 버튼을 고정시키기 위해서는 `Footer` 위로 새 요소를 덮어야할텐데, 
main의 height가 하단에 고정된 요소의 높이에 따라 달라지는 게 아니라 `h-[calc(100vh-70px)]`로 고정되어 있어서 
70px 이상의 높이를 가지는 요소가 하단에 고정된 경우
토스트가 가려지는 이슈가 발생하지 않을까요??

제가 지성님 말씀을 제대로 이해하지 못하고 있는 걸까요??ㅜㅜ